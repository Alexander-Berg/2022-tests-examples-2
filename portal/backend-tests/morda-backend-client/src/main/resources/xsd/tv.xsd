<?xml version="1.0" encoding="UTF-8"?>
<xsd:schema xmlns:xsd="http://www.w3.org/2001/XMLSchema"
            targetNamespace="urn:tv.beans.mordabackend.autotests.yandex.ru"
            xmlns:ns="urn:tv.beans.mordabackend.autotests.yandex.ru"
            elementFormDefault="qualified">

    <xsd:complexType name="tv">
        <xsd:sequence>
            <xsd:element name="exp" type="xsd:string"/>
            <xsd:element name="show" type="xsd:int"/>
            <xsd:element name="af" type="xsd:int"/>
            <xsd:element name="processed" type="xsd:int"/>
            <xsd:element name="weekend_mode" type="xsd:int"/>
            <xsd:element name="evening_mode" type="xsd:int"/>
            <xsd:element name="href" type="xsd:string"/>
            <xsd:element name="programms" type="ns:tvEvent" minOccurs="0" maxOccurs="unbounded"/>
            <xsd:element name="tabs" type="ns:tvTab" minOccurs="0" maxOccurs="unbounded"/>
            <xsd:element name="groupping" type="xsd:int"/>
            <xsd:element name="announces" type="ns:tvAnnounce" minOccurs="0" maxOccurs="unbounded"/>
        </xsd:sequence>
    </xsd:complexType>

    <xsd:complexType name="tvTab">
        <xsd:sequence>
            <xsd:element name="href" type="xsd:string"/>
            <xsd:element name="type" type="xsd:string"/>
            <xsd:element name="channel" type="xsd:int"/>
            <xsd:element name="channel_id" type="xsd:string"/>
            <xsd:element name="chname" type="xsd:string"/>
            <xsd:element name="programms" type="ns:tvEvent" minOccurs="0" maxOccurs="unbounded"/>
        </xsd:sequence>
    </xsd:complexType>

    <xsd:complexType name="tvEvent">
        <xsd:sequence>
            <xsd:element name="raw_href" type="xsd:string"/>
            <xsd:element name="time" type="xsd:string"/>
            <xsd:element name="name" type="xsd:string"/>
            <xsd:element name="channel" type="xsd:string"/>
            <xsd:element name="full" type="xsd:string"/>
            <xsd:element name="ch_href" type="xsd:string"/>
            <xsd:element name="ch_id" type="xsd:string"/>
            <xsd:element name="subtitle" type="xsd:string"/>
            <xsd:element name="href" type="xsd:string"/>
            <xsd:element name="chf" type="xsd:string"/>
            <xsd:element name="type" type="xsd:string"/>
            <xsd:element name="title" type="xsd:string"/>
            <xsd:element name="i" type="xsd:int"/>
        </xsd:sequence>
    </xsd:complexType>

    <xsd:complexType name="tvAnnounce">
        <xsd:sequence>
            <xsd:element name="lang" type="xsd:string"/>
            <xsd:element name="filter" type="xsd:string"/>
            <xsd:element name="content" type="xsd:string"/>
            <xsd:element name="tanker" type="xsd:string"/>
            <xsd:element name="domain" type="xsd:string"/>
            <xsd:element name="to" type="xsd:string"/>
            <xsd:element name="text" type="xsd:string"/>
            <xsd:element name="geos" type="xsd:string"/>
            <xsd:element name="from" type="xsd:string"/>
            <xsd:element name="url" type="xsd:string"/>
            <xsd:element name="id" type="xsd:string"/>
        </xsd:sequence>
    </xsd:complexType>

</xsd:schema>

