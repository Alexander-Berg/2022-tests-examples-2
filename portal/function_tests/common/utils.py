# -*- coding: utf-8 -*-
import logging

import allure
from hamcrest import assert_that

logger = logging.getLogger(__name__)


def get_field(json_data, path, safe=False):
    if isinstance(path, basestring):
        path_list = path.split('.')
    elif isinstance(path, list):
        path_list = path
    else:
        if not safe:
            raise AttributeError('Invalid path for json: "{}"'.format(path))
        else:
            return None

    result = json_data
    current_path = []

    for p in path_list:
        current_path.append(p)
        if isinstance(result, dict) and p in result and result[p] is not None:
                result = result[p]
        elif isinstance(result, (list, tuple)):
            if p.isdigit() and int(p) < len(result):
                result = result[int(p)]
        else:
            logger.error('Failed to get path `{}` from json'.format('.'.join(current_path)))
            if not safe:
                raise KeyError('Failed to get path `{}` from json'.format('.'.join(current_path)))
            else:
                return None

    return result


def check_field(json_data, path, matcher, safe=False):
    with allure.step('Check field {}: {}'.format(path, matcher)):
        value = get_field(json_data, path, safe)

        assert_that(value, matcher, 'Invalid "{}" value'.format(path))

def delete(hashmap, key):
    value = None
    if isinstance(hashmap, dict) and key in hashmap:
        value = hashmap[key]
        del hashmap[key]
    return value

def dump_obj(obj):
    for attr in dir(obj):
        if hasattr( obj, attr ):
            print "obj.%s = %s" % (attr, getattr(obj, attr))

def parse_set_cookie(set_cookie_str = ""):
    # yandexuid=964148831652799476; Expires=Fri, 14-May-2032 14:57:55 GMT; Domain=.yandex.ru; Path=/; Secure; SameSite=None, yuidss=964148831652799476; Expires=Fri, 14-May-2032 14:57:55 GMT; Domain=.yandex.ru; Path=/; Secure; SameSite=None
    week = {
        "sun": 1,
        "mon": 1,
        "tue": 1,
        "wed": 1,
        "thu": 1,
        "fri": 1,
        "sat": 1,
    }
    set_cookie = []
    start = 0
    pos = set_cookie_str.find(",")
    while pos != -1:
        if pos - start > 3 and set_cookie_str[pos-4] == "=" and set_cookie_str[pos-3:pos].lower() in week:
            pos = set_cookie_str.find(",", pos + 1)
        else:
            set_cookie.append(set_cookie_str[start:pos])
            start = pos + 1
            if start >= len(set_cookie_str):
                break
            pos = set_cookie_str.find(",", start)
    if start < len(set_cookie_str):
        set_cookie.append(set_cookie_str[start:])

    return set_cookie

class MadmMockError(Exception):
    pass

def check_madm_error(data):
    if data is not None:
        try:
            assert 'madm_mock_error' not in data
        except AssertionError:
            raise MadmMockError('madm_mock_error={}'.format(data['madm_mock_error']))
