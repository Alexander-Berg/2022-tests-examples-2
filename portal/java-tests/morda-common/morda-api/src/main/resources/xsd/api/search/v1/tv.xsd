<?xml version="1.0" encoding="UTF-8"?>
<xsd:schema xmlns:xsd="http://www.w3.org/2001/XMLSchema"
            xmlns:jaxb="http://java.sun.com/xml/ns/jaxb"
            jaxb:version="2.1"
            targetNamespace="urn:tv.v1.search.api.beans.morda.autotests.yandex.ru"
            xmlns:ns="urn:tv.v1.search.api.beans.morda.autotests.yandex.ru"
            xmlns:block="urn:v1.search.api.beans.morda.autotests.yandex.ru"
            xmlns:annox="http://annox.dev.java.net"
            jaxb:extensionBindingPrefixes="annox"
            elementFormDefault="qualified">

    <xsd:import schemaLocation="block.xsd"
                namespace="urn:v1.search.api.beans.morda.autotests.yandex.ru"/>

    <xsd:complexType name="TvApiV1">
        <xsd:annotation>
            <xsd:appinfo>
                <annox:annotate target="class">
                    @ru.yandex.qatools.processors.matcher.gen.annotations.GenerateMatcher
                </annox:annotate>
            </xsd:appinfo>
        </xsd:annotation>
        <xsd:complexContent>
            <xsd:extension base="block:Block">
                <xsd:sequence>
                    <xsd:element name="data" type="ns:TvApiV1Data"/>
                </xsd:sequence>
            </xsd:extension>
        </xsd:complexContent>
    </xsd:complexType>

    <xsd:complexType name="TvApiV1Data">
        <xsd:annotation>
            <xsd:appinfo>
                <annox:annotate target="class">
                    @ru.yandex.qatools.processors.matcher.gen.annotations.GenerateMatcher
                </annox:annotate>
            </xsd:appinfo>
        </xsd:annotation>
        <xsd:sequence>
            <xsd:element name="geo" type="xsd:int"/>
            <xsd:element name="url" type="xsd:string"/>
            <xsd:element name="tab" type="ns:TvApiV1Tab" minOccurs="0" maxOccurs="unbounded"/>
        </xsd:sequence>
    </xsd:complexType>

    <xsd:complexType name="TvApiV1Tab">
        <xsd:annotation>
            <xsd:appinfo>
                <annox:annotate target="class">
                    @ru.yandex.qatools.processors.matcher.gen.annotations.GenerateMatcher
                </annox:annotate>
            </xsd:appinfo>
        </xsd:annotation>
        <xsd:sequence>
            <xsd:element name="url" type="xsd:string"/>
            <xsd:element name="title" type="xsd:string"/>
            <xsd:element name="type" type="xsd:string"/>
            <xsd:element name="program" type="ns:TvApiV1Program" minOccurs="0" maxOccurs="unbounded"/>
        </xsd:sequence>
    </xsd:complexType>

    <xsd:complexType name="TvApiV1Program">
        <xsd:annotation>
            <xsd:appinfo>
                <annox:annotate target="class">
                    @ru.yandex.qatools.processors.matcher.gen.annotations.GenerateMatcher
                </annox:annotate>
            </xsd:appinfo>
        </xsd:annotation>
        <xsd:sequence>
            <xsd:element name="type" type="xsd:string"/>
            <xsd:element name="program_id" type="xsd:string"/>
            <xsd:element name="event_id" type="xsd:string"/>
            <xsd:element name="ttl" type="xsd:int"/>
            <xsd:element name="time" type="xsd:string"/>
            <xsd:element name="channel" type="xsd:string"/>
            <xsd:element name="url" type="xsd:string"/>
            <xsd:element name="title" type="xsd:string"/>
        </xsd:sequence>
    </xsd:complexType>

</xsd:schema>

