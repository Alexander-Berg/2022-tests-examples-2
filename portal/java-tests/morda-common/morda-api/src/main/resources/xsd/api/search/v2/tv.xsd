<?xml version="1.0" encoding="UTF-8"?>
<xsd:schema xmlns:xsd="http://www.w3.org/2001/XMLSchema"
            xmlns:jaxb="http://java.sun.com/xml/ns/jaxb"
            jaxb:version="2.1"
            targetNamespace="urn:tv.v2.search.api.beans.morda.autotests.yandex.ru"
            xmlns:ns="urn:tv.v2.search.api.beans.morda.autotests.yandex.ru"
            xmlns:block="urn:v2.search.api.beans.morda.autotests.yandex.ru"
            xmlns:annox="http://annox.dev.java.net"
            jaxb:extensionBindingPrefixes="annox"
            elementFormDefault="qualified">

    <xsd:import schemaLocation="block.xsd"
                namespace="urn:v2.search.api.beans.morda.autotests.yandex.ru"/>

    <xsd:complexType name="TvApiV2">
        <xsd:complexContent>
            <xsd:extension base="block:Block">
                <xsd:sequence>
                    <xsd:element name="data" type="ns:TvApiV2Data"/>
                </xsd:sequence>
            </xsd:extension>
        </xsd:complexContent>
    </xsd:complexType>

    <xsd:complexType name="TvApiV2Data">
        <xsd:sequence>
            <xsd:element name="geo" type="xsd:int"/>
            <xsd:element name="url" type="xsd:string"/>
            <xsd:element name="tab" type="ns:TvApiV2Tab" minOccurs="0" maxOccurs="unbounded"/>
        </xsd:sequence>
    </xsd:complexType>

    <xsd:complexType name="TvApiV2Tab">
        <xsd:sequence>
            <xsd:element name="url" type="xsd:string"/>
            <xsd:element name="title" type="xsd:string"/>
            <xsd:element name="type" type="xsd:string"/>
            <xsd:element name="program" type="ns:TvApiV2Program" minOccurs="0" maxOccurs="unbounded"/>
        </xsd:sequence>
    </xsd:complexType>

    <xsd:complexType name="TvApiV2Program">
        <xsd:sequence>
            <xsd:element name="type" type="xsd:string"/>
            <xsd:element name="program_id" type="xsd:string"/>
            <xsd:element name="event_id" type="xsd:string"/>
            <xsd:element name="ttl" type="xsd:int"/>
            <xsd:element name="time" type="xsd:string"/>
            <xsd:element name="channel" type="xsd:string"/>
            <xsd:element name="url" type="xsd:string"/>
            <xsd:element name="title" type="xsd:string"/>
        </xsd:sequence>
    </xsd:complexType>

</xsd:schema>

