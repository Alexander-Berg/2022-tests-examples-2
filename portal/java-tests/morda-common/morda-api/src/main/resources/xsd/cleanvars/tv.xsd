<?xml version="1.0" encoding="UTF-8"?>
<xsd:schema xmlns:xsd="http://www.w3.org/2001/XMLSchema"
            targetNamespace="urn:tv.cleanvars.beans.morda.autotests.yandex.ru"
            xmlns:ns="urn:tv.cleanvars.beans.morda.autotests.yandex.ru"
            xmlns:annox="http://annox.dev.java.net"
            xmlns:jaxb="http://java.sun.com/xml/ns/jaxb"
            jaxb:version="2.1"
            jaxb:extensionBindingPrefixes="annox"
            elementFormDefault="qualified">

    <xsd:complexType name="tv">
        <xsd:annotation>
            <xsd:appinfo>
                <annox:annotate target="class">
                    @ru.yandex.qatools.processors.matcher.gen.annotations.GenerateMatcher
                </annox:annotate>
            </xsd:appinfo>
        </xsd:annotation>

        <xsd:sequence>
            <xsd:element name="exp" type="xsd:string"/>
            <xsd:element name="show" type="xsd:int"/>
            <xsd:element name="geo" type="xsd:int"/>
            <xsd:element name="af" type="xsd:int"/>
            <xsd:element name="processed" type="xsd:int"/>
            <xsd:element name="weekend_mode" type="xsd:int"/>
            <xsd:element name="evening_mode" type="xsd:int"/>
            <xsd:element name="href" type="xsd:string"/>
            <xsd:element name="programms" type="ns:tvEvent" minOccurs="0" maxOccurs="unbounded"/>
            <xsd:element name="tabs" type="ns:tvTab" minOccurs="0" maxOccurs="unbounded"/>
            <xsd:element name="groupping" type="xsd:int"/>
            <xsd:element name="announces" type="ns:tvAnnounce" minOccurs="0" maxOccurs="unbounded"/>
        </xsd:sequence>
    </xsd:complexType>

    <xsd:complexType name="tvTab">
        <xsd:annotation>
            <xsd:appinfo>
                <annox:annotate target="class">
                    @ru.yandex.qatools.processors.matcher.gen.annotations.GenerateMatcher
                </annox:annotate>
            </xsd:appinfo>
        </xsd:annotation>


        <xsd:sequence>
            <xsd:element name="title" type="xsd:string"/>
            <xsd:element name="href" type="xsd:string"/>
            <xsd:element name="type" type="xsd:string"/>
            <xsd:element name="channel" type="xsd:int"/>
            <xsd:element name="channel_id" type="xsd:string"/>
            <xsd:element name="chname" type="xsd:string"/>
            <xsd:element name="programms" type="ns:tvEvent" minOccurs="0" maxOccurs="unbounded"/>
        </xsd:sequence>
    </xsd:complexType>

    <xsd:complexType name="tvEvent">
        <xsd:annotation>
            <xsd:appinfo>
                <annox:annotate target="class">
                    @ru.yandex.qatools.processors.matcher.gen.annotations.GenerateMatcher
                </annox:annotate>
            </xsd:appinfo>
        </xsd:annotation>

        <xsd:sequence>
            <xsd:element name="separator" type="xsd:int"/>
            <xsd:element name="ch_href" type="xsd:string"/>
            <xsd:element name="ch_id" type="xsd:string"/>
            <xsd:element name="channel" type="xsd:string"/>
            <xsd:element name="end_ts" type="xsd:long"/>
            <xsd:element name="full" type="xsd:string"/>
            <xsd:element name="href" type="xsd:string"/>
            <xsd:element name="name" type="xsd:string"/>
            <xsd:element name="pr_id" type="xsd:string"/>
            <xsd:element name="program_id" type="xsd:string"/>
            <xsd:element name="start_ts" type="xsd:long"/>
            <xsd:element name="time" type="xsd:string"/>
            <xsd:element name="title" type="xsd:string"/>
            <xsd:element name="type" type="xsd:string"/>
            <xsd:element name="raw_href" type="xsd:string"/>
            <xsd:element name="subtitle" type="xsd:string"/>
            <xsd:element name="chf" type="xsd:string"/>
            <xsd:element name="i" type="xsd:int"/>
        </xsd:sequence>
    </xsd:complexType>

    <xsd:complexType name="tvAnnounce">
        <xsd:annotation>
            <xsd:appinfo>
                <annox:annotate target="class">
                    @ru.yandex.qatools.processors.matcher.gen.annotations.GenerateMatcher
                </annox:annotate>
            </xsd:appinfo>
        </xsd:annotation>

        <xsd:sequence>
            <xsd:element name="lang" type="xsd:string"/>
            <xsd:element name="filter" type="xsd:string"/>
            <xsd:element name="content" type="xsd:string"/>
            <xsd:element name="tanker" type="xsd:string"/>
            <xsd:element name="domain" type="xsd:string"/>
            <xsd:element name="to" type="xsd:string"/>
            <xsd:element name="text" type="xsd:string"/>
            <xsd:element name="geos" type="xsd:string"/>
            <xsd:element name="from" type="xsd:string"/>
            <xsd:element name="url" type="xsd:string"/>
            <xsd:element name="id" type="xsd:string"/>
        </xsd:sequence>
    </xsd:complexType>

</xsd:schema>

