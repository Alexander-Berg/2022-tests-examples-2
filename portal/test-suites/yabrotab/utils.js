'use strict';

function setWidth (width, cols, ratio) {
    var page = document.querySelector('.b-page'),
        pageStyle = page.style,
        GAP = 8;

    const oneItem = (width - (cols - 1) * GAP) / cols,
        wide = cols > 4;

    page.classList.toggle('b-page_width_wide', wide);
    page.classList.toggle('b-page_width_normal', !wide);
    pageStyle.setProperty('--page-width', `${width}px`);
    pageStyle.setProperty('--item-ratio', ratio);
    pageStyle.setProperty('--item-width', `${oneItem}px`);
    pageStyle.setProperty('--cols-count', cols);
    if (window.broZen) {
        window.broZen.setWidth(+width);
    }
}

async function waitForPageReady() {
    await this.browser.yaInBrowser(async () => {
        await new Promise(resolve => {
            if (document.readyState === 'complete') {
                return resolve();
            }
            window.addEventListener('load', resolve);
        });

        await new Promise(resolve => {
            if (window.pageStatus.visible) {
                return resolve();
            }
            window.pageStatus.addEventListener('page-visible-changed', resolve);
        });
    });

    await this.browser.$("[class*='b-page_scrollable']").then(elem => elem.waitForExist({
        timeout: 5000
    }));
}

async function isBlocksRefreshedAfter(commands) {
    const TIMEOUT = 5000;
    const startTs = await this.browser.yaInBrowser(function getStartTs() {
        return document.querySelector('.main').dataset.ts;
    });

    try {
        await commands.call(this);
    } catch (e) {
        e.message = `isBlocksRefreshedAfter: ошибка при выполнении комманд: ${e.message}`;
    }
    await this.browser
        .yaInBrowser(function checkEndTs(startTs, TIMEOUT) {
            const ts = document.querySelector('.main').dataset.ts;

            if (ts === startTs) {
                return;
            }

            return new Promise((resolve, reject) => {
                const observer = new MutationObserver(list => list.map(onMutationItem));

                function onMutationItem(item) {
                    if (item.addedNodes && Array.prototype.some.call(item.addedNodes, node => node.closest('.main'))) {
                        const endTs = document.querySelector('.main').dataset.ts;

                        if (endTs !== startTs) {
                            observer.disconnect();
                            resolve();
                        }
                    }
                }

                setTimeout(() => {
                    observer.disconnect();
                    reject(new Error(`Блоки не обновлись за ${TIMEOUT}мс`));
                }, TIMEOUT);

                observer.observe(document.querySelector('.main').parentNode, {childList: true});
            });
        }, startTs, TIMEOUT);

}

async function isBlocksNotRefreshed() {
    const TIMEOUT = 5000;
    const {error} = await this.browser
        .executeAsync(function getTimestaps(TIMEOUT, done) {
            const observer = new MutationObserver(list => list.map(onMutationItem));

            function onMutationItem(item) {
                if (item.addedNodes && Array.prototype.some.call(item.addedNodes, node => node.closest('.main'))) {
                    observer.disconnect();
                    done({
                        error: 'Начался рефреш'
                    });
                }
            }
            setTimeout(() => {
                observer.disconnect();
                done({});
            }, TIMEOUT);

            observer.observe(document.querySelector('.main').parentNode, {childList: true});
        }, TIMEOUT);

    if (error) {
        throw new Error(error);
    }
}

module.exports = {
    setWidth,
    waitForPageReady,
    isBlocksRefreshedAfter,
    isBlocksNotRefreshed,
    ignore: [
        '.news',
        '.weather2',
        '.traffic',
        '.feed'
    ],
    hide: [
        '.dev-panel',
        '.panel-item',
        '.zen-app__logo',
        '.banner',
        // Новости могут быть двустрочными, растягивая контент
        '.list__item'
    ]
};
