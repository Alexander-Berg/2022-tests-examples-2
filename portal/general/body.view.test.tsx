/* eslint-disable @typescript-eslint/ban-ts-comment */

import { execView } from '@lib/views/execView';
import { mockView, restoreViews } from '@lib/views/mock';
import { mockReq } from '@lib/views/mockReq';
import { DepsTree } from '@lib/utils/depsTree';
// import { Search3 } from '@block/search3/search3.view';
// import { Mtabs } from '@block/mtabs/mtabs.view';
import { Mlogo } from '@block/mlogo/mlogo.view';
import { BannerDebug } from '@block/banner-debug/banner-debug.view';
import { Body__blocksBottom, Body__blocksTail } from '@block/body/body.view';
import { Search3 } from '@block/search3/search3.view';
import { News } from '@block/news/news.view';
// @ts-ignore
import { Banner } from '@block/banner/banner.view';
// @ts-ignore
import { Tv } from '@block/tv/tv.view';

// todo move to imports

describe('body', () => {
    describe('split-inline', () => {
        let getSpy: jest.SpyInstance;
        let postponeSpy: jest.SpyInstance;
        const createOverride = () => {
            const depsTree = new DepsTree({}, new Proxy({}, {
                get: (target, key) => key
            }));
            getSpy = jest.spyOn(depsTree, 'get');
            postponeSpy = jest.spyOn(depsTree, 'postpone');
            return {
                depsTree,
                blocks_layout: [
                    'news',
                    'tv'
                ],
                options: {
                    disable_plus_wallet_touch: '1'
                }
            };
        };

        let origFlag: boolean;
        beforeEach(() => {
            mockView(BannerDebug, () => '');
            mockView(Mlogo, () => 'Logo');
            mockView(Search3, () => '');

            mockView(Banner, () => '');
            mockView(Body__blocksBottom, () => []);
            mockView(Body__blocksTail, () => []);

            mockView(News, () => 'News');
            mockView(Tv, () => 'Tv');

            origFlag = home.env.inlineCss;
            home.env.inlineCss = true;
        });

        afterEach(() => {
            restoreViews();

            home.env.inlineCss = origFlag;
        });

        test('process without group exp', () => {
            const req = mockReq({}, createOverride());

            expect(execView('Body__content', {}, req)).toMatchSnapshot();
            expect(getSpy.mock.calls).toMatchSnapshot();
            expect(postponeSpy.mock.calls).toMatchSnapshot();
        });
    });
});
