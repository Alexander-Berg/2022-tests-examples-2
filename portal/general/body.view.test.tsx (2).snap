// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`body split-inline process without group exp 1`] = `"<body class='i-bem body_lite_yes' data-bem='{\\"i-global\\":{}}'><style  >body__wrapperbody__before</style><div class='body__wrapper'><div class='content i-bem'><div  class=\\"body__before\\"></div><style  >Body__blocksHead</style><style  >body__blocks-head</style><div class='theader'><div class='theader__personal'>Logo<div class='theader__personal-right'><a class=\\"home-link2 home-link2_color_inherit\\" href=\\"https://undefined/auth?msid=1.1.1.1&origin=home_yaru_touch_new\\" data-statlog=\\"header.login\\"><div class='theader__login'>Войти</div></a></div></div><div class='theader__search'><svg class=\\"theader__search-icon\\" width=\\"180\\" height=\\"72\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M47.048 21.385h6.389v27.103h-6.389V21.385zm-19.467 0h6.426v27.103h-6.426V21.385zm2.609 3.894v-3.894h19.353v3.894H30.19zM69.447 45.2c1.159 0 2.154-.228 2.986-.681.831-.479 1.461-1.197 1.89-2.155.454-.982.68-2.242.68-3.78 0-1.487-.227-2.709-.68-3.666-.428-.983-1.058-1.701-1.89-2.155-.832-.479-1.827-.718-2.986-.718-1.11 0-2.092.24-2.949.718-.831.454-1.474 1.172-1.928 2.155-.453.957-.68 2.18-.68 3.666 0 1.537.227 2.785.68 3.742.454.958 1.097 1.676 1.928 2.155.857.479 1.84.718 2.949.718zm0 3.666c-2.319 0-4.373-.403-6.162-1.21-1.764-.806-3.15-1.965-4.158-3.477-.983-1.538-1.474-3.402-1.474-5.595 0-2.192.504-4.044 1.512-5.556 1.008-1.537 2.394-2.697 4.158-3.478 1.764-.806 3.805-1.21 6.124-1.21 2.343 0 4.397.404 6.161 1.21 1.79.781 3.175 1.94 4.158 3.478 1.008 1.512 1.512 3.364 1.512 5.556 0 2.193-.504 4.057-1.512 5.595-1.008 1.537-2.394 2.709-4.158 3.515-1.764.781-3.818 1.172-6.161 1.172zM86.028 48.488l1.852-2.382 12.815-17.388h4.611l-1.625 1.966-12.966 17.804h-4.687zm5.103-19.77v14.78l-.794 4.99h-5.254v-19.77h6.048zm9.866 0h5.292v19.77h-6.086v-15.46l.794-4.31zM122.354 48.866c-2.52 0-4.7-.403-6.54-1.21-1.814-.831-3.225-2.003-4.233-3.515-.983-1.537-1.475-3.39-1.475-5.557 0-2.142.492-3.969 1.475-5.48 1.008-1.538 2.419-2.71 4.233-3.516 1.815-.832 3.957-1.248 6.426-1.248 1.109 0 2.117.076 3.024.227.907.126 1.726.328 2.457.605a9.785 9.785 0 0 1 2.004.945v3.704a12.2 12.2 0 0 0-2.155-.944 9.98 9.98 0 0 0-2.155-.53c-.73-.1-1.537-.15-2.419-.15-1.487 0-2.721.226-3.704.68a4.663 4.663 0 0 0-2.23 2.116c-.504.958-.756 2.155-.756 3.591 0 1.462.252 2.671.756 3.63a5.122 5.122 0 0 0 2.268 2.116c.982.453 2.179.68 3.591.68 1.411 0 2.633-.126 3.666-.378a14.234 14.234 0 0 0 3.138-1.285v3.667a9.667 9.667 0 0 1-2.042 1.02c-.73.278-1.537.48-2.419.605-.882.151-1.852.227-2.91.227zM133.144 28.718h6.086v19.77h-6.086v-19.77zm21.47 19.77h-7.144l-8.316-9.752 9.374-10.017h5.406l-9.715 10.206.492-1.966 9.903 11.529z\\" fill=\\"currentColor\\"/></svg></div></div><div class='content__informers-bg'><div class='informers'><span class=\\"home-link2 informers__all home-link2_color_gray home-link2_hover_red\\" data-statlog=\\"informers.all\\">Ещё</span></div></div><style  >news</style>News<script  >(document.querySelector(\\".content__informers-bg\\")||{}).offsetHeight</script><style  >tv</style>Tv<style  >zen-inserts</style><div class='zen-inserts'></div><div class='tail'></div></div></div><div  class=\\"lite-switcher-touch lite-switcher-touch_enabled_yes i-bem\\" data-bem=\\"{&quot;lite-switcher-touch&quot;:{}}\\"><button class=\\"button lite-switcher-touch__inner button_theme_lite-switcher i-bem\\" data-bem=\\"{&quot;button&quot;:{}}\\" aria-pressed=\\"true\\" role=\\"button\\" type=\\"button\\" data-statlog=\\"lite-switcher.switch.disable\\"><span class=\\"button__text\\">Старый дизайн <span class='lite-switcher-touch__switch'></span></span></button></div></body>"`;

exports[`body split-inline process without group exp 2`] = `
Array [
  Array [
    Array [
      "body__wrapper",
      "body__before",
    ],
  ],
  Array [
    "body__blocks-head",
  ],
  Array [
    Array [
      "Body__blocksHead",
    ],
  ],
  Array [
    Array [
      "news",
    ],
  ],
  Array [
    Array [
      "tv",
    ],
  ],
  Array [
    "zen-inserts",
  ],
]
`;

exports[`body split-inline process without group exp 3`] = `Array []`;
